test/Microsoft.AspNet.BasicMiddleware.Tests/OverrideHeaderMIddlewareTest.cs

```diff
@@ -0,0 +1,206 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Net.Http;
+using System.Threading.Tasks;
+using Microsoft.AspNet.Builder;
+using Microsoft.AspNet.TestHost;
+using Xunit;
+
+namespace Microsoft.AspNet.BasicMiddleware
+{
+    public class OverrideMiddlewareHeaderTests
+    {
+        [Fact]
+        public async Task OverrideHeaderMiddleware_XForwardEnabled()
```

Comment:
Please read the [engineering guidelines](https://github.com/aspnet/Home/wiki/Engineering-guidelines#unit-test-method-naming) on good unit test naming patterns.